torch/csrc/cuda/comm.cpp

```diff
@@ -111,4 +118,89 @@ tensor_list2d broadcast_coalesced(TensorList tensors, IntList devices, size_t bu
   return outputs;
 }
 
-}}
+std::vector<at::Tensor> scatter(
+    const at::Tensor& tensor,
+    at::IntList devices,
+    const at::optional<std::vector<int64_t>>& chunk_sizes,
+    int64_t dim,
+    const at::optional<std::vector<THCStream*>>& streams) {
+  std::vector<at::Tensor> chunks;
+  if (!chunk_sizes) {
+    chunks = tensor.chunk(/*chunks=*/devices.size(), /*dim=*/dim);
+  } else {
+    const int64_t chunk_size_sum =
+        std::accumulate(chunk_sizes->begin(), chunk_sizes->end(), 0);
+    // clang-format off
+    AT_CHECK(
+      chunk_size_sum == tensor.size(dim),
+      "given chunk sizes don't sum up to the tensor's size ",
+      "(sum(chunk_sizes) == ", chunk_size_sum,
+      ", but expected ", tensor.size(dim), ")");
+    // clang-format on
```

Comment:
Is clang-format really messing this thing up? It's annoying to have meaningless comments in the code.

Reply:
Got rid of the comments